src/main/java/com/godaddy/commerce/migrator/handlers/helpers/BatcherHelper.java

```diff
@@ -1,81 +1,81 @@
 package com.godaddy.commerce.migrator.handlers.helpers;
 
+import static com.godaddy.commerce.migrator.models.Constants.ENTITLEMENTS;
+import static com.godaddy.commerce.migrator.models.Constants.LEGACY_FULFILLMENT;
+import static com.godaddy.commerce.migrator.models.Constants.SHOPPERS;
 import static java.lang.String.format;
 
 import com.godaddy.commerce.core.errors.ServiceException;
 import com.godaddy.commerce.core.http.HttpVerbResult;
-import com.godaddy.commerce.core.http.HttpVerbWrapper;
-import com.godaddy.commerce.core.http.config.ConnectionProperty;
-import com.godaddy.commerce.core.http.config.EcommerceHost;
+import com.godaddy.commerce.core.http.HttpVerbWrapperContext;
+import com.godaddy.commerce.migrator.models.BackfillEntitlementRequest;
 import com.godaddy.commerce.migrator.models.Shopper;
+import com.godaddy.commerce.services.vertx.util.jwttoken.SSOJWTHelper;
 import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 
 import org.apache.http.HttpStatus;
 
 import java.net.InetAddress;
 import java.net.UnknownHostException;
-import java.time.Duration;
 import java.util.UUID;
 
 @Slf4j
+@RequiredArgsConstructor
 public class BatcherHelper {
 
-  static final String SHOPPERS_CONNECTION = "SHOPPERS_PCI";
-  static final String ENTITLEMENTS_CONNECTION = "ENTITLEMENTS";
   private static String HOST_ADDRESS;
-  private static final ConnectionProperty connectionProperty = EcommerceHost.INSTANCE
-    .connectionProperties()
-    .get(SHOPPERS_CONNECTION);
-  private String shopperCert = connectionProperty.cert();
-  private String shopperUrl = connectionProperty.url();
-  private String endpoint = "/v1/shoppers/%s";
-  private String legacyUrlBase = EcommerceHost.INSTANCE.connectionProperties().get("LEGACY-FULFILLMENT")
-    .url();
-  private String legacyUrl = "%s/v2/customers/%s/legacy-fulfillment/processLegacyEntitlement";
-  private ConnectionProperty entConn = EcommerceHost.INSTANCE.connectionProperties().get(ENTITLEMENTS_CONNECTION);
-  private final String entitlementsBase = entConn.url();
-  private String getEntitlementEndpoint = "%s/v2/customers/%s/entitlements/%s";
+  private final SSOJWTHelper ssojwtHelper;
 
   static {
     try {
       HOST_ADDRESS = InetAddress.getLocalHost().getHostAddress(); //NOSONAR
-    }
-    catch (UnknownHostException e) {
+    } catch (UnknownHostException e) {
       throw new ServiceException("Could not get Host Address", e);
     }
   }
 
-  public BatcherHelper() {
-
+  public HttpVerbResult<JsonObject> processLegacyEntitlement(final BackfillEntitlementRequest backfillEntitlementRequest, final String customerId, final RoutingContext routingContext) {
+    final String legacyUri = "%s/v2/customers/%s/legacy-fulfillment/processLegacyEntitlement";
+    final HttpVerbWrapperContext wrapperContext = new HttpVerbWrapperContext(
+      LEGACY_FULFILLMENT,
+      routingContext
```

Comment:
> [<img alt="mtbennett" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.secureserver.net/mtbennett) **Authored by [mtbennett](https://github.secureserver.net/mtbennett)**
_<time datetime="2021-02-03T17:43:54Z" title="Wednesday, February 3rd 2021, 10:43:54 am -07:00">Feb 3, 2021</time>_

---


`LEGACY_FULFILLMENT` is cluster-local so this needs `PREFER_LOCAL_DNS` option. This will resolve the discussion above.

```suggestion
      routingContext,
      HttpVerbOption.PREFER_LOCAL_DNS
```